Automattic/jetpack
# Stats: explicitly allow only the designated API with blog token

#### Changes proposed in this Pull Request:
- Explicitly allow only the designated API forwarding with blog token.
- Added functions to `WPOM_Stats` to retrieve `highlights`, `insights` and `streak` 
- Changed to use `WPCOM_Stats` to retrieve stats for the controller

The context of the backend is to fully satisfy the needs of the frontend. The frontend isn’t purposefully built for Jetpack but is actually a Calypso app requesting ‘public-api’, which is the cause of the site Ids in the route.
We want to quickly spin up the backend so that the whole application works in Jetpack and then we’ll iterate and improve from there.

So I’m happy to leave few stubs there for now as long as they don’t cause any concerns because It’ll take significantly longer to optimise the Calypso app - we’ll need to refactor several places to remove these api calls, and make the app slimmer too and I want to deal with these on the next stage.

#### Other information:


#### Jetpack product discussion
p1669283843924099-slack-C0438NHCLSY

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Open a site with Jetpack plugin or simply spin up a JN site
- Open `/wp-admin/admin.php?page=jetpack#/traffic?term=jetpack%20stats`
- Expand the Jetpack Stats section
- Open browser console and run `document.querySelector('#jp-settings-site-stats').querySelectorAll('.jp-form-fieldset')[1].setAttribute('style','color:red')`
- Turn on the new Stats experience (color red toggle)
- Open `/wp-admin/admin.php?page=stats`
- Ensure there is data fill in the page
